### PR TITLE
Ensure QList items are added and sorted.

### PR DESCRIPTION
**Pull request by chris2oph**
This change ensures that the RearrangeListTestCase tests now pass. It was necessary to ensure the stored order array was being sized correctly as well as ensuring that the items were being added to the list for QT.

### CODE REVIEW

File: src/common/rearrangectrl.cpp [hunk 1]
```diff
@@ -70,7 +70,8 @@ bool wxRearrangeList::Create(wxWindow *parent,
     }
 
     // do create the real control
-    m_order.reserve(count);
+    m_order = order;
```
Comment (vadz) [inline src/common/rearrangectrl.cpp]: Could you please explain why was this changed?

File: src/qt/listbox.cpp [hunk 2]
```diff
@@ -106,14 +106,17 @@ bool wxListBox::Create(wxWindow *parent, wxWindowID id,
...
-            const wxArrayString& WXUNUSED(choices),
+            const wxArrayString& choices,
             long style,
             const wxValidator& validator,
             const wxString& name)
 {
     Init();
     m_qtWindow = m_qtListWidget = new wxQtListWidget( parent, this );
 
+    for (size_t i = 0; i < choices.size(); ++i)
```
Comment (vadz) [inline src/qt/listbox.cpp]: I don't mind doing it like this, but I think it could be more efficient to construct a `QStringList` and call `addItems()` once, instead of calling `addItem()` in a loop many times.

File: src/common/rearrangectrl.cpp (hunk 1 above)
Comment (chris2oph) [reply src/common/rearrangectrl.cpp]: Shortly after the reservation of memory in the `m_order` member at line 73, there is a loop that attempts to reorder the member. However attempting to assign to the member in the loop in this fashion: `m_order[n] = order[n]` caused the program to throw an exception. I believe this is because `reserve()` only sets the capacity and not the size of the array, so when we try to access an element in the array there are none there.

File: src/qt/listbox.cpp (hunk 2 above)
Comment (chris2oph) [reply src/qt/listbox.cpp]: An excellent suggestion, I'll get this applied and committed ASAP. Many thanks!

Comment (chris2oph) [reply src/qt/listbox.cpp]: I am struggling to see a way I can do this change without having to loop over the `choices` array in some fashion. I am unable to convert the `wxString`s to `QString`s outside of a loop, so to apply your suggested change in this instance would mean we still have the same loop (albeit with different contents). Perhaps the string conversion in the loop along with the call to `addLines()` outside the loop would be more efficient but I am not certain is a worthy enough saving to warrant a further change.

Comment (vadz) [reply src/qt/listbox.cpp]: Sorry for being unclear, there will still definitely be a loop. The idea is just it's often more efficient to add many items to the control at once rather than adding them in a loop because you only pay the "item added" overhead (a control typically needs to do something more than just adding the string to an internal list) once instead of N times in this case. So "loop to create QStringList" + "a single call to addItems()" should be more efficient, especially for big amounts (and there can be hundreds of them in some listboxes), than "loop calling addItem".

Comment (chris2oph) [reply src/qt/listbox.cpp]: Excellent, I appreciate the clarification. I'll get this change committed soon!

File: src/common/rearrangectrl.cpp (hunk 1 above)
Comment (vadz) [reply src/common/rearrangectrl.cpp]: Yes, you're absolutely right, the current code is invalid and doesn't work at all if `order` argument is not empty. Apparently this got broken in 7e1b64a2eb6  -- @a-wi do you remember why was this particular change done? I understand the other changes in this commit, but not what was wrong with `m_order = order`.

In any case, we need to either revert this part of that commit or replace `reserve()` with `resize()`. The current version is just broken, while the new version doesn't make much sense: why assign the whole array first and then assign it again in the loop, the second assignment is useless (and removing it would be effectively the same as reverting part of 7e1b64a2eb6).

So I won't apply this part of the changes now, but we do need to fix it.

Comment (chris2oph) [reply src/common/rearrangectrl.cpp]: No problem @vadz  . I have implemented your suggested change of using `resize()` rather than `reserve()`.
